packages/teleterm/src/ui/ClusterRemove/ClusterRemove.tsx

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright 2021 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect } from 'react';
+import DialogConfirmation, {
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+} from 'design/DialogConfirmation';
+import * as Alerts from 'design/Alert';
+import { Text, ButtonPrimary, ButtonSecondary } from 'design';
+import { useAppContext } from 'teleterm/ui/appContextProvider';
+import useAsync from 'teleterm/ui/useAsync';
+
+export default function ClusterRemove({
+  onClose,
+  clusterTitle,
+  clusterUri,
+}: Props) {
+  const ctx = useAppContext();
+  const [{ status, statusText }, run] = useAsync(() => {
+    return ctx.serviceClusters.removeCluster(clusterUri);
+  });
+
+  useEffect(() => {
```

Comment:
it's better to have a separate hook called `useClusterRemove` that takes care of the state which we can inject via 

```tsx
import useClusterRemove, { State } from './useClusterRemove';

export default function Container() {
  const ctx = useAppContext();
  const state = useClusterRemove(ctx);
  return <ClusterRemove {...state} />;
}

export function ClusterRemove(props: <State>) {
}
```

This will make it easier to add to the storybook and add snapshot tests in the future.

Reply:
Done